imap_processing/idex/idex_l1a.py

```diff
@@ -228,6 +230,39 @@ def _read_waveform_bits(waveform_raw: str, high_sample: bool = True) -> list[int
     return ints
 
 
+def calculate_idex_epoch_time(
+    shcoarse_time: Union[float, np.ndarray], shfine_time: Union[float, np.ndarray]
+) -> npt.NDArray[np.int64]:
+    """
+    Calculate the epoch time from the FPGA header time variables.
+
+    We are given the MET seconds, we need to convert it to nanoseconds in j2000. IDEX
+    epoch is calculated with shcoarse and shfine time values. The shcoarse time counts
+    the number of whole seconds elapsed since the epoch (Jan 1st 2010), while shfine
+    time counts the number of additional 20-microsecond intervals beyond the whole
+    seconds. Together, these time measurements establish when a dust event took place.
+
+    Parameters
+    ----------
+    shcoarse_time : float, numpy.ndarray
+        The coarse time value from the FPGA header. Number of seconds since epoch.
+    shfine_time : float, numpy.ndarray
+        The fine time value from the FPGA header. Number of 20 microsecond "ticks" since
+         the last second.
+
+    Returns
+    -------
+    numpy.ndarray[numpy.int64]
+        The mission elapsed time converted to nanoseconds since the J2000 epoch
+        in the terrestrial time (TT) timescale.
+    """
+    # Number of microseconds since the last second
+    microseconds_since_last_second = 20 * shfine_time
+    # Get met time in seconds including shfine
+    met = shcoarse_time + microseconds_since_last_second * 1e-6
```

Comment:
Definitely a nit here, but I find the split out into another quantity slightly more confusing than just inline multiplying by the 20 microseconds. Just suggesting it here in case that reads better to you too but it is fine as-is as well.
```suggestion
    # Get met time in seconds including shfine which is the number of 20 microseconds ticks
    met = shcoarse_time + shfine_time * 20e-6
```

Reply:
I think I agree with you on this one.